scripts/ci/prepare_environment.py

```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Callable, Tuple, List
+
+from utils.config import contracts, get_deployer_account
+from utils.voting import bake_vote_items
+from utils.dual_governance import process_pending_proposals
+from utils.evm_script import encode_call_script
+from utils.import_current_votes import get_vote_script_files, get_upgrade_script_files
+from utils.mainnet_fork import pass_and_exec_dao_vote
+
+def main():
+    process_pending_proposals()
+    execute_votings()
+
+def execute_votings():
+    votings_in_flight = retrieve_votings_in_flight()
+    vote_script = retrieve_vote_script()
+
+    votings_to_execute = list(votings_in_flight)
+
+    if vote_script:
+        start_vote, get_vote_items = vote_script
+
+        is_already_in_flight = any(
+            is_vote_script_equal_to_voting_in_flight(vote_id, get_vote_items)
+            for vote_id in votings_in_flight
+        )
+
+        if not is_already_in_flight:
+            print("Starting a new voting from script...")
+            new_vote_id, _ = start_vote(tx_params={"from": get_deployer_account()}, silent=True)
+            votings_to_execute.append(new_vote_id)
+
+    if not votings_to_execute:
+        print("No votings to execute.")
+        return
+
+    print(f"Passing and enacting votings: {votings_to_execute}")
+    for vote_id in votings_to_execute:
+        pass_and_exec_dao_vote(vote_id=vote_id)
+
+def is_vote_script_equal_to_voting_in_flight(
+    voting_id: int, get_vote_items: Callable[[], Tuple[List, List]]
+):
+    onchain_vote_script = contracts.voting.getVote(voting_id)["script"]
+
+    vote_desc_items, call_script_items = get_vote_items()
+    local_vote_items = bake_vote_items(list(vote_desc_items), list(call_script_items))
+    local_evm_script = encode_call_script(local_vote_items.values())
+
+    return local_evm_script == onchain_vote_script
+
+def retrieve_votings_in_flight() -> List[int]:
+    votings_in_flight = []
+    last_vote_id = contracts.voting.votesLength() - 1
+
+    while last_vote_id >= 0:
+        if contracts.voting.getVotePhase(last_vote_id) == 2:
```

Comment:
Ensure this helper works with votes that have not reached quorum and are in the objection phase.